xdsl/utils/mlir_lexer.py

```diff
@@ -64,26 +64,58 @@ def string_contents(self):
 
     @property
     def bytes_contents(self) -> bytes:
+        # NOTE: This function is used to convert the string literal into a bytes object and
+        # handle escape sequences. There are two types of escape sequences we process:
+        # 1. Known escape sequences like \n, \t, \\, and \"
+        # 2. Hexadecimal escape sequences like \x00, \x0E, \00, \0E, etc.
```

Comment:
Let's add a doc comment here:
```suggestion
        """
        Convert the string literal into a bytes object and handle escape sequences. 
        There are two types of escape sequences we process:
        1. Known escape sequences like \n, \t, \\, and \"
        2. Hexadecimal escape sequences like \x00, \x0E, \00, \0E, etc.
        """
```

Reply:
Done by https://github.com/xdslproject/xdsl/pull/4834/commits/dcf7f34756732fa8b42b66f37cb09cafe183bee8